upup/models/cloudup/resources/addons/authentication.aws/k8s-1.12.yaml.template

```
@@ -145,16 +149,20 @@ spec:
             port: 21362
             scheme: HTTPS
         volumeMounts:
+        {{- if contains "MountedFile" .Authentication.Aws.BackendMode }}
```

I think all of the conditions around volume mounts need to also evaluate to true if the backend mode is not specified in order to preserve the default behavior.

I'm not positive on the syntax but it should be something similar to:
```suggestion
        {{- if or (not .Authentication.Aws.BackendMode) (contains "MountedFile" .Authentication.Aws.BackendMode) }}
```

---

ah, good catch, thanks. I'll add that this evening when I have some more time to work on it.